CMakeLists.txt

```
@@ -1,13 +1,19 @@
 cmake_minimum_required(VERSION 3.10)
-project(vikunja)
+project(vikunja
+  VERSION      0.1.0
+  DESCRIPTION  "Primitives for Alpaka"
+  HOMEPAGE_URL "https://github.com/ComputationalRadiationPhysics/vikunja"
+  LANGUAGES    CXX)
 
 set(CMAKE_CXX_STANDARD 14)
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra")
```

MSVC screams.

---

No problem, I also scream, if I use the MSVC XD Joke aside, I didn't look on this line. In general, the library is not tested with Windows or MacOSX. But before fixing this line, I should look after the reason. But maybe in another PR. Jiri's students need this stuff.

---

Is there a reason why we are setting`CMAKE_CXX_FLAGS` directly? That is usually considered bad style since this is a flag that is set by the user (if required). The only exception I know of is the (outdated) `FindCUDA.cmake` script which doesn't play nice with CMake targets. If vikunja isn't using this I suggest the following change:

```suggestion
if(MSVC)
    target_compile_options(vikunja INTERFACE "/W4")
else()
    target_compile_options(vikunja INTERFACE "-Wall -Wextra -pedantic")
endif()
```

---

It is an artifact, from the old CMake 2.x style without targets, which I forgot. Maybe the heading of the PR is wrong. My target was to make vikunja nice usable in other project. So didn't care about the sections, which are not direct required for my task. But I think, I should change my opinion ;-)

You suggestion sounds good. But I want to check also, if it necessary in general.

---

I had a mistake in my mind. I thought it reduce the amount of warnings. I used @j-stephan approach and added a option to make it optional.